pkg/sentry/platform/ring0/kernel_amd64.go

```diff
@@ -18,13 +18,27 @@ package ring0
 
 import (
 	"encoding/binary"
+	"reflect"
+
+	"gvisor.dev/gvisor/pkg/usermem"
 )
 
 // init initializes architecture-specific state.
-func (k *Kernel) init(opts KernelOpts) {
+func (k *Kernel) init(opts KernelOpts, maxCPUs int) {
 	// Save the root page tables.
 	k.PageTables = opts.PageTables
 
+	k.cpuEntryArea = &cpuEntryArea{}
+	if len(k.cpuEntryArea.cpuEntries) < maxCPUs {
+		panic("Fixed-defined _RING0_MAX_CPUS is less than maxCPUs")
+	}
+	areaStart := reflect.ValueOf(&k.cpuEntryArea.alignStart[0]).Pointer()
+	areaEnd := reflect.ValueOf(&k.cpuEntryArea.alignEnd[0]).Pointer()
+	alignedStart, _ := usermem.Addr(areaStart).RoundUp()
+	alignedEnd, _ := usermem.Addr(areaEnd).RoundUp()
+	k.CpuEntryAreaStart = uintptr(alignedStart)
```

Comment:
Noted elsewhere: these should just propagated via method called on the Kernel object here, instead of indirectly via opts (confusing and requires additional documentation). E.g.

```go
func (k *Kernel) VirtualRegions() map[usermem.Addr]uint64 {
   r := make(map[usermem.Addr]uint64)
   addr := reflect.ValueOf(&k.cpuEntryArea[0]).Pointer()
   r[addr] = len(k.cpuEntryArea)
   return r
}
```

Reply:
it is done as you suggested (function name and function body are different)